plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java

```diff
@@ -1799,6 +1799,27 @@ private void doTestBucketedTableValidation(HiveStorageFormat storageFormat, Sche
                     .hasErrorCode(HIVE_INVALID_BUCKET_FILES)
                     .hasMessageMatching("Hive table is corrupt\\. File '.*/000002_0_.*' is for bucket 2, but contains a row for bucket 5.");
         }
+
+        // mark table as created by Spark
+        try (Transaction transaction = newTransaction()) {
+            ConnectorSession session = newSession();
+            PrincipalPrivileges principalPrivileges = testingPrincipalPrivilege(session);
+            Table oldTable = transaction.getMetastore().getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName()).get();
+            Table.Builder newTable = Table.builder(oldTable).setParameter(HiveMetadata.SPARK_TABLE_PROVIDER_KEY, "parquet");
+            transaction.getMetastore().replaceTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName(), newTable.build(), principalPrivileges);
+            transaction.commit();
+        }
+
+        // read succeeds on tables marked with Spark-specific property
```

Comment:
Can we add another test for a bucketed table written by Spark but with the table properties removed and then asserting that the reads fail? Or is there such case already covered in the sampling bucketing validator cursor?

Reply:
Sure, I have a Spark-generated TPCH nation table in ORC format somewhere. I'll see how easy it is to add it the tests.

Reply:
Turns out this is more difficult than I originally estimated. The `AbstractTestHive` framework only allows creating empty tables and injecting rows from a `MaterializedResult` with Presto, I don't see any way to import an existing set of bucketed ORC files. I've also considered writing a product test, with an actual Spark instance creating a Hive-compatible table, but it turned out Spark doesn't support bucketing in this mode.

Reply:
In some tests, we copy a prepared file into table's location.
see eg https://github.com/trinodb/trino/blob/072ac47331d96a8bee79bc83a971b11406ca69dc/plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveIntegrationSmokeTest.java#L3573

Reply:
Note that you might need a local Spark instance anyway to prepare the test files (one-time activity) by having Spark write a bucketed dataframe into ORC files (no need to write a Hive table).

Also this better captures the actual workflow people have when they face issues querying Spark bucketed tables from Trino. They use Spark to write files into a location which happens to be a Hive table location.

Reply:
> They use Spark to write files into a location which happens to be a Hive table location.


in such a case current fix will not kick in, will it?

Reply:
> in such a case current fix will not kick in, will it?

Yes. No way to know if files were bucketed. But that can happen without Spark in the picture too - with people copying files into a bucketed table without understanding the file naming nomenclature.

Reply:
I agree. And we would need to ask user for help (aka session property) in such a case.
Then the question is -- which problem are we trying to solve here?

Reply:
Spark does allow writing to bucketed tables - IIRC newer (very recent) versions prevent writing to a bucketed Hive table but it can be overridden.

Reply:
> Note that you might need a local Spark instance anyway to prepare the test files (one-time activity) by having Spark write a bucketed dataframe into ORC files (no need to write a Hive table).

I was thinking about preparing those ORC files once and adding them to resources. That way I'd avoid running Spark.

Reply:
Turns out the initial approach with `AbstractTestHive` was the right way to go, I was just missing a couple of pieces. Thanks for your suggestions.